packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako

```diff
@@ -187,3 +194,18 @@ class ${class_name}:
 
         for response in self._get_stub().Measure(request):
             yield self._deserialize_response(response)
+
+    def register_pin_map(self, pin_map_path: str) -> str:
+        """Registers the pin map with the pin map service.
+        
+        Args:
+            pin_map_path: Absolute path of the pin map file.
+
+        Returns:
+            Registered pin map id.
+        """
+        pin_map_client = PinMapClient()
+        global _pin_map_path
```

Comment:
Instead of keeping it as a global variable, could you please add it as a parameter for the class?

Reply:
Moved `pin_map_path` and `sites` as class members.